spec/services/patient_import/spreadsheet_transformer_spec.rb

```diff
@@ -215,4 +215,48 @@
       expect(patient[:prescription_drugs].first).not_to have_key(:rxnorm_code)
     end
   end
+
+  context "for abbreviated genders" do
+    it "recognizes 'm'" do
+      params = PatientImport::SpreadsheetTransformer.call(data, facility: facility)
+
+      patient = params.find { |p| p[:patient][:full_name] == "Gender Abbrev M" }.deep_symbolize_keys
+
+      expect(patient[:patient][:gender]).to eq("male")
+    end
+
+    it "recognizes 'f'" do
+      params = PatientImport::SpreadsheetTransformer.call(data, facility: facility)
+
+      patient = params.find { |p| p[:patient][:full_name] == "Gender Abbrev F" }.deep_symbolize_keys
+
+      expect(patient[:patient][:gender]).to eq("female")
+    end
+
+    it "recognizes 't'" do
+      params = PatientImport::SpreadsheetTransformer.call(data, facility: facility)
+
+      patient = params.find { |p| p[:patient][:full_name] == "Gender Abbrev T" }.deep_symbolize_keys
+
+      expect(patient[:patient][:gender]).to eq("transgender")
+    end
+  end
+
+  context "for abbreviated died? status" do
```

Comment:
These tests don't exactly match the code. The code converts "y" and "yes" to dead and everything else to active. We should add a test for "yes" and perhaps make it clear in tests that it doesn't matter if it says "N" or "Alive" or "Deceased", they all translate to "active"

Reply:
Great call. I'll add some more test data that contains all sorts of values 👍